sql/catalyst/src/main/java/org/apache/spark/sql/connector/write/RequiresDistributionAndOrdering.java

```diff
@@ -47,6 +47,22 @@ public interface RequiresDistributionAndOrdering extends Write {
    */
   Distribution requiredDistribution();
 
+  /**
+   * Returns if the distribution required by this write is strictly required or best effort only.
+   * <p>
+   * If true, Spark will strictly distribute incoming records across partitions to satisfy
```

Comment:
It's probably better to not mention the internal optimizations so clearly. Maybe just say
```
If true, Spark will strictly distribute incoming records across partitions to satisfy
the required distribution before passing the records to the data source table on write.
Otherwise, Spark may apply certain optimizations to speed up the query but break
the distribution requirement.
```